src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterDocumentRewriter.cs

```diff
@@ -0,0 +1,695 @@
+                                argumentListSyntax, invocation, validParameters, parameterName, newMethodIdentifier, insertionIndex, cancellationToken);
+                        });
+                    }
+                }
+
+                // If you are at the original document, then also introduce the new method and introduce the parameter.
+                if (currentDocument.Id == _originalDocument.Id)
+                {
+                    var newMethodNode = _actionKind is IntroduceParameterCodeActionKind.Trampoline
+                        ? await ExtractMethodAsync(validParameters, newMethodIdentifier, generator, cancellationToken).ConfigureAwait(false)
+                        : await GenerateNewMethodOverloadAsync(insertionIndex, generator, cancellationToken).ConfigureAwait(false);
+                    editor.InsertBefore(_containerMethod, newMethodNode);
+
+                    await UpdateExpressionInOriginalFunctionAsync(editor, cancellationToken).ConfigureAwait(false);
+                    var parameterType = await GetTypeOfExpressionAsync(cancellationToken).ConfigureAwait(false);
+                    var parameter = generator.ParameterDeclaration(parameterName, generator.TypeExpression(parameterType));
+                    editor.InsertParameter(_containerMethod, insertionIndex, parameter);
+                }
+
+                return editor.GetChangedRoot();
+
+                // Adds an argument which is an invocation of the newly created method to the callsites
+                // of the method invocations where a parameter was added.
+                // Example:
+                // public void M(int x, int y)
+                // {
+                //     int f = [|x * y|];
+                //     Console.WriteLine(f);
+                // }
+                // 
+                // public void InvokeMethod()
+                // {
+                //     M(5, 6);
+                // }
+                //
+                // ---------------------------------------------------->
+                // 
+                // public int GetF(int x, int y)
+                // {
+                //     return x * y;
+                // }
+                // 
+                // public void M(int x, int y)
+                // {
+                //     int f = x * y;
+                //     Console.WriteLine(f);
+                // }
+                //
+                // public void InvokeMethod()
+                // {
+                //     M(5, 6, GetF(5, 6)); // This is the generated invocation which is a new argument at the call site
+                // }
+                SyntaxNode GenerateNewArgumentListSyntaxForTrampoline(Compilation compilation, ISyntaxFactsService syntaxFacts, ISemanticFactsService semanticFacts,
+                    SemanticModel invocationSemanticModel, SyntaxGenerator generator, Dictionary<IParameterSymbol, int> parameterToArgumentMap,
```

Comment:
since you don't support cross language rewriting, you coudl pull syntaxfact/semanticfacts/generator up to be members on the containing type.  that would simplify this.

Reply:
done